### PR TITLE
Move Linux-X86-Clang targets to CircleCI

### PR DESCRIPTION
**Pull request by xiaobai**
Let's move them :D

### CODE REVIEW

**Comment by xiaobai** [comment]
I found 32bit libstdc++ in `/usr/lib32` on my docker image, we need to point it there probably.

File: .circleci/config.yml [hunk 1]
```diff
@@ -90,6 +90,14 @@ jobs:
     environment:
       TARGET: Linux-X86
       PLATFORM: 1
+  Linux-X86-Clang:
+    <<: *build_and_test_defaults
+  Linux-X86-Clang-platform:
+    <<: *build_and_test_defaults
+    environment:
+      TARGET: Linux-X86-Clang
+      RELEASE: 1
```
Comment (sas) [inline .circleci/config.yml]: You can remove this line. We don't support `RELEASE=1` on CircleCI. I'm not sure if it's useful, but if it is, I'll add it later on.

Comment (xiaobai) [reply .circleci/config.yml]: Ah, alright. Sounds good.

**Comment by xiaobai** [comment]
Yeah, I will move those in a separate PR.

**Comment by xiaobai** [comment]
Once the effects of #584 kick in, I'll re-run the builds here and my expectation is that ds2 should build and the tests should run. I can't guarantee all tests will pass though. :P

**Pull request merged**